servatrice/src/servatrice_database_interface.h

```diff
@@ -71,13 +76,13 @@ public slots:
     bool checkUserIsBanned(const QString &ipAddress, const QString &userName, const QString &clientId, QString &banReason, int &banSecondsRemaining);
     int checkNumberOfUserAccounts(const QString &email);
     bool registerUser(const QString &userName, const QString &realName, ServerInfo_User_Gender const &gender,
-        const QString &password, const QString &emailAddress, const QString &country, QString &token, bool active = false);
+        const QString &password, const QString &emailAddress, const QString &country, QString &token, const QString &clientid, bool active = false);
     bool activateUser(const QString &userName, const QString &token);
     void updateUsersClientID(const QString &userName, const QString &userClientID);
     void updateUsersLastLoginData(const QString &userName, const QString &clientVersion);
     void logMessage(const int senderId, const QString &senderName, const QString &senderIp, const QString &logMessage,
         LogMessage_TargetType targetType, const int targetId, const QString &targetName);
-    bool changeUserPassword(const QString &user, const QString &oldPassword, const QString &newPassword);
+    bool changeUserPassword(const QString &user, const QString &oldPassword, const QString &newPassword, bool force = false);
```

Comment:
The original use of the function was to change a users password ,  you would pass in the old password and new password.  The function would check if the old password matched and if so would reset the password, otherwise it wouldn't reset the user password.  To avoid building pretty much as similar function I simply appended the bool value (force) to this function and if the force option is passed the old password check is bypassed and the password is simply reset.  Since we dont know the users account old password during the password reset process we have no way to check against it :)